---
title: "eslint/no-fallthrough | The JavaScript Oxidation Compiler"
source_url: "https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough"
fetched_at: "2025-12-31T10:44:14.234719+00:00"
---



Are you an LLM? You can read better optimized documentation at /docs/guide/usage/linter/rules/eslint/no-fallthrough.md for this page in Markdown format

# eslint/no-fallthrough Pedantic [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#eslint-no-fallthrough)

üöß An auto-fix is planned for this rule, but not implemented at this time.

### What it does [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#what-it-does)

Disallow fallthrough of `case` statements

This rule is aimed at eliminating unintentional fallthrough of one case to the other. As such, it flags any fallthrough scenarios that are not marked by a comment.

### Why is this bad? [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#why-is-this-bad)

The switch statement in JavaScript is one of the more error-prone constructs of the language thanks in part to the ability to ‚Äúfall through‚Äù from one case to the next. For example:

js

```
switch (foo) {
  case 1:
    doSomething();

  case 2:
    doSomethingElse();
}
```

In this example, if `foo` is `1`, then execution will flow through both cases, as the first falls through to the second. You can prevent this by using `break`, as in this example:

js

```
switch (foo) {
  case 1:
    doSomething();
    break;

  case 2:
    doSomethingElse();
}
```

That works fine when you don‚Äôt want a fallthrough, but what if the fallthrough is intentional, there is no way to indicate that in the language. It‚Äôs considered a best practice to always indicate when a fallthrough is intentional using a comment which matches the `/falls?\s?through/i`` regular expression but isn‚Äôt a directive:

js

```
switch (foo) {
  case 1:
    doSomething();
  // falls through

  case 2:
    doSomethingElse();
}

switch (foo) {
  case 1:
    doSomething();
  // fall through

  case 2:
    doSomethingElse();
}

switch (foo) {
  case 1:
    doSomething();
  // fallsthrough

  case 2:
    doSomethingElse();
}

switch (foo) {
  case 1: {
    doSomething();
    // falls through
  }

  case 2: {
    doSomethingElse();
  }
}
```

In this example, there is no confusion as to the expected behavior. It is clear that the first case is meant to fall through to the second case.

### Examples [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#examples)

Examples of **incorrect** code for this rule:

js

```
switch (foo) {
  case 1:
    doSomething();

  case 2:
    doSomething();
}
```

Examples of **correct** code for this rule:

js

```
switch (foo) {
  case 1:
    doSomething();
    break;

  case 2:
    doSomething();
}

function bar(foo) {
  switch (foo) {
    case 1:
      doSomething();
      return;

    case 2:
      doSomething();
  }
}

switch (foo) {
  case 1:
    doSomething();
    throw new Error("Boo!");

  case 2:
    doSomething();
}

switch (foo) {
  case 1:
  case 2:
    doSomething();
}

switch (foo) {
  case 1:
  case 2:
    doSomething();
}

switch (foo) {
  case 1:
    doSomething();
  // falls through

  case 2:
    doSomething();
}

switch (foo) {
  case 1: {
    doSomething();
    // falls through
  }

  case 2: {
    doSomethingElse();
  }
}
```

Note that the last case statement in these examples does not cause a warning because there is nothing to fall through into.

## Configuration [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#configuration)

This rule accepts a configuration object with the following properties:

### allowEmptyCase [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#allowemptycase)

type: `boolean`

default: `false`

Whether to allow empty case clauses to fall through.

### commentPattern [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#commentpattern)

type: `string | null`

Custom regex pattern to match fallthrough comments.

### reportUnusedFallthroughComment [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#reportunusedfallthroughcomment)

type: `boolean`

default: `false`

Whether to report unused fallthrough comments.

## How to use [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#how-to-use)

To **enable** this rule using the config file or in the CLI, you can use:

Config (.oxlintrc.json)CLI

json

```
{
  "rules": {
    "no-fallthrough": "error"
  }
}
```

bash

```
oxlint --deny no-fallthrough
```

## References [‚Äã](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-fallthrough.html#references)

* [Rule Source](https://github.com/oxc-project/oxc/blob/1bf0ffc0f6859c90409a9701e62e8957ef1286cc/crates/oxc_linter/src/rules/eslint/no_fallthrough.rs)
