---
title: "eslint/no-return-assign | The JavaScript Oxidation Compiler"
source_url: "https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign"
fetched_at: "2025-12-31T10:44:14.234719+00:00"
---



Are you an LLM? You can read better optimized documentation at /docs/guide/usage/linter/rules/eslint/no-return-assign.md for this page in Markdown format

# eslint/no-return-assign Style [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#eslint-no-return-assign)

ðŸš§ An auto-fix is planned for this rule, but not implemented at this time.

### What it does [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#what-it-does)

Disallows assignment operators in return statements.

### Why is this bad? [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#why-is-this-bad)

Assignment is allowed by js in return expressions, but usually, an expression with only one equal sign is intended to be a comparison. However, because of the missing equal sign, this turns to assignment, which is valid js code Because of this ambiguity, itâ€™s considered a best practice to not use assignment in return statements.

### Examples [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#examples)

Examples of **incorrect** code for this rule:

js

```
() => (a = b);
function x() {
  return (a = b);
}
```

Examples of **correct** code for this rule:

js

```
() => (a = b);
function x() {
  var result = (a = b);
  return result;
}
```

## Configuration [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#configuration)

This rule accepts one of the following string values:

### `"always"` [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#always)

Disallow all assignments in return statements.

### `"except-parens"` [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#except-parens)

Allow assignments in return statements only if they are enclosed in parentheses. This is the default mode.

## How to use [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#how-to-use)

To **enable** this rule using the config file or in the CLI, you can use:

Config (.oxlintrc.json)CLI

json

```
{
  "rules": {
    "no-return-assign": "error"
  }
}
```

bash

```
oxlint --deny no-return-assign
```

## References [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/eslint/no-return-assign.html#references)

* [Rule Source](https://github.com/oxc-project/oxc/blob/1bf0ffc0f6859c90409a9701e62e8957ef1286cc/crates/oxc_linter/src/rules/eslint/no_return_assign.rs)
