---
title: "promise/prefer-catch | The JavaScript Oxidation Compiler"
source_url: "https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch"
fetched_at: "2025-12-31T10:44:14.234719+00:00"
---



Are you an LLM? You can read better optimized documentation at /docs/guide/usage/linter/rules/promise/prefer-catch.md for this page in Markdown format

# promise/prefer-catch Style [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch.html#promise-prefer-catch)

ðŸš§ An auto-fix is planned for this rule, but not implemented at this time.

### What it does [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch.html#what-it-does)

Prefer `catch` to `then(a, b)` and `then(null, b)`. This rule disallows the passing of an argument into the second parameter of `then` calls for handling promise errors.

### Why is this bad? [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch.html#why-is-this-bad)

A `then` call with two arguments can make it more difficult to recognize that a catch error handler is present. Another issue with using the second argument in `then` calls is that the ordering of promise error handling is less obvious.

For example on first glance it may appear that `prom.then(fn1, fn2)` is equivalent to `prom.then(fn1).catch(fn2)`. However they aren't equivalent. In fact `prom.catch(fn2).then(fn1)` is the equivalent. This kind of confusion is a good reason for preferring explicit `catch` calls over passing an argument to the second parameter of `then` calls.

### Examples [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch.html#examples)

Examples of **incorrect** code for this rule:

js

```
prom.then(fn1, fn2);

prom.then(null, fn2);
```

Examples of **correct** code for this rule:

js

```
prom.catch(fn2).then(fn1);

prom.catch(fn2);
```

## How to use [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch.html#how-to-use)

To **enable** this rule using the config file or in the CLI, you can use:

Config (.oxlintrc.json)CLI

json

```
{
  "plugins": ["promise"],
  "rules": {
    "promise/prefer-catch": "error"
  }
}
```

bash

```
oxlint --deny promise/prefer-catch --promise-plugin
```

## References [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/promise/prefer-catch.html#references)

* [Rule Source](https://github.com/oxc-project/oxc/blob/1bf0ffc0f6859c90409a9701e62e8957ef1286cc/crates/oxc_linter/src/rules/promise/prefer_catch.rs)
