---
title: "unicorn/prefer-dom-node-dataset | The JavaScript Oxidation Compiler"
source_url: "https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset"
fetched_at: "2025-12-31T10:44:14.234719+00:00"
---



Are you an LLM? You can read better optimized documentation at /docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.md for this page in Markdown format

# unicorn/prefer-dom-node-dataset Pedantic [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.html#unicorn-prefer-dom-node-dataset)

ðŸš§ An auto-fix is planned for this rule, but not implemented at this time.

### What it does [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.html#what-it-does)

Use [`.dataset`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) on DOM elements over `getAttribute(â€¦)`, `.setAttribute(â€¦)`, `.removeAttribute(â€¦)` and `.hasAttribute(â€¦)`.

### Why is this bad? [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.html#why-is-this-bad)

The `dataset` property is a map of strings that contains all the `data-*` attributes from the element. It is a convenient way to access all of them at once.

### Examples [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.html#examples)

Examples of **incorrect** code for this rule:

javascript

```
element.setAttribute("data-unicorn", "ðŸ¦„");
```

Examples of **correct** code for this rule:

javascript

```
element.dataset.unicorn = "ðŸ¦„";
```

## How to use [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.html#how-to-use)

To **enable** this rule using the config file or in the CLI, you can use:

Config (.oxlintrc.json)CLI

json

```
{
  "rules": {
    "unicorn/prefer-dom-node-dataset": "error"
  }
}
```

bash

```
oxlint --deny unicorn/prefer-dom-node-dataset
```

## References [â€‹](https://oxc.rs/docs/guide/usage/linter/rules/unicorn/prefer-dom-node-dataset.html#references)

* [Rule Source](https://github.com/oxc-project/oxc/blob/1bf0ffc0f6859c90409a9701e62e8957ef1286cc/crates/oxc_linter/src/rules/unicorn/prefer_dom_node_dataset.rs)
